SemanticSearch2.md

```diff
@@ -0,0 +1,320 @@
+		- weight (float)
+		- type (string: skill, job_title, keyword, company, school, location, company_geography)
+		- entity_node (object of type-specific attributes for type: location and company_geography)
+		- relationships (map of canonical string to list of relationship objects)
+			- occupations (for type: job_title, skill, company, keyword)
+			- related_keywords (for type: job_title, skill, company, keyword)
+			- textkernel_related_keywords (for type: job_title, skill, company, keyword)
+			- skills (for type: job_title, skill, company, keyword)
+			- job_titles (for type: job_title, skill, company, keyword)
+			- job_level (for type: job_title, skill, company, keyword)
+			- education_level (for type:school)
+	- versions (map of relationship string to version string)
+
+
+2.0 response:
+```
+{
+	"data" : {
+		"parsed_input" : {
+			"input" : "Java developer,Careerbuilder",
+			"parsed_fragments" : [
+				"java developer",
+				"careerbuilder"
+			],
+			"input_to_extracted_keywords" : {
+				"Careerbuilder" : "careerbuilder",
+				"Java developer" : "java developer"
+			}
+		},
+		"extracted_keywords" : [
+			{
+				"name" : "java developer",
+				"relationships" : {
+					"skills" : [
+						{
+							"weight" : 1,
+							"name" : "Java (Programming Language)",
+							"id" : "KS120076FGP5WGWYMP0F"
+						},
+						{
+							"name" : "Hibernate (Java)",
+							"id" : "KS124PR62MV42B5C9S9F",
+							"weight" : 0.93154
+						},
+						{
+							"id" : "KS123KG6DL8N3D5ZW036",
+							"name" : "Java Platform Enterprise Edition (J2EE)",
+							"weight" : 0.92962
+						}
+					],
+					"job_titles" : [
+						{
+							"id" : "15.2",
+							"name" : "Java Developer",
+							"weight" : 1
+						}
+					],
+					"job_level" : [
+						{
+							"name" : "Experienced (non-Manager)",
+							"id" : "3",
+							"weight" : 1
+						}
+					],
+					"text_kernel_related_keywords" : [
+						{
+							"name" : "Java Architect",
+							"weight" : 1
+						},
+						{
+							"name" : "J2EE",
+							"weight" : 1
+						},
+						{
+							"weight" : 1,
+							"name" : "Java Programmer"
+						}
+					],
+					"occupations" : [],
+					"related_keywords" : []
+				},
+				"type" : "job_title",
+				"weight" : 100
+			},
+			{
+				"name" : "careerbuilder",
+				"type" : "company",
+				"weight" : 100
+				"relationships" : {},
+			},
+			{
+				"type" : "job_level",
+				"name" : "EXPERIENCED_NON_MANAGER"
+			},
+			{
+				"type" : "location",
+				"entity_node" : {
+					"address" : "Atlanta, Georgia",
+					"region" : "Georgia"
+				}
+			},
+			{
+				"name" : "Java developer",
+				"type" : "job_title"
+			}
+		],
+		"versions": {
+			"extracted_keywords": "ExtractedKeywords",
+			"job_titles": "CaroteneV3",
+			"skills": "SkillsV4",
+			"related_keywords": "RelatedSearchTermsV1",
+			"text_kernel_related_keywords": "TextKernelRelatedSearchTerms",
+			"custom_keywords": "CustomKeywords"
+		}
+	}
+}
+```
+# Error Handling
```

Comment:
I think we could remove this section. All the invalid input will receive Bad Request 400. Missing parameters and empty requests are bad requests, too actually. Actually it may be a better idea to remove the documentation for the error messages since we may update that and forgot to update accordingly in the documentation.

What we need to ensure is to provide detailed information about how to use the parameters, I saw you already did. users who don't supply the required parameters in the correct way should expect an bad request.